src/schema/README.md

```diff
@@ -483,35 +624,229 @@ Specifically, files in the first group may have `task`, `run`, `processing`, and
-[#1096](https://github.com/bids-standard/bids-specification/pull/1096)
+An additional check will be required to assert that both are not present, but
+these tables may be combined for rendering purposes.
+
+Here we present an example rule in `rules.tabular_data.eeg`:
+
+```YAML
+EEGChannels:
+  selectors:
+  - datatype == "eeg"
+  - suffix == "channels"
+  - extension == ".tsv"
+  initial_columns:
+  - name__channels
+  - type__eeg_channels
+  - units
+  columns:
+    name__channels: required
+    type__eeg_channels: required
+    units: required
+    description: optional
+    sampling_frequency: optional
+    reference: optional
+    low_cutoff: optional
+    high_cutoff: optional
+    notch: optional
+    status: optional
+    status_descriptions: optional
+  additional_columns: allowed_if_defined
+```
+
+### Checks
+
+`rules.checks` can contain more complex rules. Structurally, these are similar to sidecar rules,
+in that they have selectors. They additionally have a `checks` list, and an explicit issue.
+
+| Field       | Description                                                                                    |
+| ----------- | ---------------------------------------------------------------------------------------------- |
+| `issue`     | Issue code object (see [Issues](#issues)                                                       |
+| `selectors` | List of expressions; any evaluating false indicate rule does not apply                         |
+| `checks`    | List of expressions; any evaluating false indicate rule is violated and issue should be raised |
+
+```YAML
+EventsMissing:
+  issue:
+    code: EVENTS_TSV_MISSING
+    message: |
+      Task scans should have a corresponding events.tsv file.
+      If this is a resting state scan you can ignore this warning or rename the task to include the word "rest".
+    level: warning # could be an error with the proper selectors, I think
+  selectors:
+    - '"task" in entities'
+    - '!matches(entities.task, "rest")'
+    - suffix != "events"
+  checks:
+    - '"events" in associations'
+```
+
+### Ordering rules
+
+-   `rules.entities` - This file contains a list of keys into `objects.entities` and
+    simply defines the order in which entities, when present, MUST appear in filenames
+
+-   `rules.common_principles` - This file contains a list of terms that appear in `objects.common_principles`
+    that determines the order they appear in the specification
+
+### One-off rules
+
+-   `rules.modalities` - The keys in this file are the modalities, the values objects with the following field:
+    | Field       | Description                           |
+    | ----------- | ------------------------------------- |
+    | `datatypes` | List of datatypes mapping to modality |
+
+-   `rules.dataset_metadata` - These are similar to `rules.sidecars.*`, for JSON files at the root level.
+    This is likely to go away in favor of other approaches.
```

Comment:
I do not like this, so I didn't bother fleshing it out. I think what we want is to fully define these documents inside the `objects.*` namespace, and find some way in the rules to say something like "a file named `dataset_description.json` contains the object `objects.metadata.DatasetDescription`".